pkg/homedir/homedir_unix.go

```diff
@@ -101,11 +101,20 @@ func isWriteableOnlyByOwner(perm os.FileMode) bool {
 	return (perm & 0o722) == 0o700
 }
 
-// GetConfigHome returns XDG_CONFIG_HOME.
-// GetConfigHome returns $HOME/.config and nil error if XDG_CONFIG_HOME is not set.
-//
-// See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
+// GetConfigHome deprecated
 func GetConfigHome() (string, error) {
+	rootlessConfigHomeDir, rootlessConfigHomeDirError = ConfigHome()
+	if rootlessConfigHomeDirError == nil {
+		_ = os.MkdirAll(rootlessConfigHomeDir, 0o700)
```

Comment:
If this function is deprecated and we intend to migrate all users, I don’t see that _changing_ the existing one to not report failures is useful: that forces us to immediately review/migrate all users, negating the benefit of the deprecation and introduction of a new name for the new version.

Reply:
Also, right now, if the directory is missing, `ConfigHome` returns `("", ENOENT)`, so AFAICS this condition does nothing.

---

Shouldn’t all these functions, with their rare corner cases we don’t encounter on our workstations, have pretty good test coverage? I still mourn for tests lost in https://github.com/containers/storage/pull/1740 .